### PR TITLE
Tom dem 226 dynamic price

### PR DESCRIPTION
**Pull request by tladendo**
fyi @vietnogi @indytechcook -- now updating prices in real-time. The `save` stuff is a little messy, but basically, the local version of `features` is different from what the API expects because it has a few extra fields that are useful to us. Anyway, this both calculates price locally and updates it based on API responses.

### CODE REVIEW

**Comment by tladendo** [comment]
P.S. I think this also fixes the 'Next' button issues.

File: src/client/app/submit-work/submit-work.service.js [hunk 1]
```diff
@@ -49,20 +51,44 @@
...
+        x.explanation = undefined;
+        x.selected = undefined;
       });
-      service.current.submitAttempted = undefined;
-      data.create('work-request', service.current)
-      .then(function(newWorkRequest) {
-        service.setCurrent(newWorkRequest);
-        promise.resolve(newWorkRequest);
+      work.submitAttempted = undefined;
+      data.create('work-request', work)
```
Comment (indytechcook) [inline src/client/app/submit-work/submit-work.service.js]: Create should only be used on new requests.  updates should use .update.

File: src/client/app/submit-work/aside/submit-work-aside.jade [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
     .type Design Only
 
-    .price $3000
+    .price ${{getPrice()}}
```
Comment (indytechcook) [inline src/client/app/submit-work/aside/submit-work-aside.jade]: Should be bind to a variable instead of a method?  Not sure about best practices, just a question.

File: src/client/app/submit-work/submit-work.service.js [hunk 3]
```diff
@@ -49,20 +51,44 @@
...
+        updatePrice();
+        promise.resolve(data);
       })
       .catch(function(e) {
         $q.reject(e);
       });
     }
 
+    function getPrice() {
+      var calcPrice = work.features.reduce(function(x, y) {
```
Comment (indytechcook) [inline src/client/app/submit-work/submit-work.service.js]: Does the backend sometimes not calculate this correctly or is this just if we haven't saved it yet?

File: src/client/app/submit-work/submit-work.service.js [hunk 4]
```diff
@@ -23,10 +23,12 @@
       setCurrent: setCurrent,
       next: next,
       save: save,
+      getPrice: getPrice,
+      updatePrice: updatePrice,
       validateName: validateName,
       validateSummary: validateSummary,
       validateUsageDescription: validateUsageDescription,
-      globalValidate: globalValidate
+      globalValidate: globalValidate,
```
Comment (indytechcook) [inline src/client/app/submit-work/submit-work.service.js]: remove comma

File: src/client/app/submit-work/submit-work.service.js (hunk 1 above)
Comment (tladendo) [inline src/client/app/submit-work/submit-work.service.js]: Since this is the same as what was already there, I think there should be another PR to fix this. Just created `DEM-251` to track this.

File: src/client/app/submit-work/submit-work.service.js (hunk 4 above)
Comment (tladendo) [inline src/client/app/submit-work/submit-work.service.js]: will do

File: src/client/app/submit-work/submit-work.service.js (hunk 3 above)
Comment (tladendo) [inline src/client/app/submit-work/submit-work.service.js]: The backend is always correct. This is just for when we haven't saved. It, e.g., makes the app more responsive when you're clicking around in the features section.

**Comment by indytechcook** [comment]
other then comments and follow up.  lgtm

File: src/client/app/submit-work/aside/submit-work-aside.jade (hunk 2 above)
Comment (tladendo) [inline src/client/app/submit-work/aside/submit-work-aside.jade]: This is a calculated value and not just a variable

**Comment by tladendo** [comment]
fyi @vietnogi i'm merging this in because prasad is asking for it. still comment on anything you see and i can fix in another pr. thx
